### PR TITLE
Feat product listing page

### LINKED ISSUE #5: Todo: Building Product Listing Page
To build the product listing page for Avavya

### PR DESCRIPTION
**Pull request by coldpigli**
1) Added Filter Context and Reducer.
2) Customized Filters according to choice.
3) Moved Product List to DB. Added custom data to all.

### CODE REVIEW

File: src/components/FilterItem.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const FilterItem = ({filter, setFilterList, filterList}) => {
+
+  const {filterDispatch} = useFilter();
+
+  const handleClick = (filter) => {
+    //to track which filters are active inorder to change active states
+    const temp = filterList.map((item)=>{
+     return (item.name===filter.name) ? ({...item, active: !item.active}): item
+    })
+    setFilterList(temp);
```
Comment (kjais1720) [inline src/components/FilterItem.jsx:12]: you can do this inside a functional update of the filterList, no need to create a `temp` then set it to filterlist

**Review by rahulcs754 — APPROVED** [review]
well, define context and reducer for your product filter.
all looks good

**Review by sadathshariff — APPROVED** [review]
The Code looks clean and neat liked how you have broken down the filters as component and reusing it

File: src/components/FilterItem.jsx (hunk 1 above)
Comment (coldpigli) [reply src/components/FilterItem.jsx:12]: Noted. Thanks for pointing it out. Will do.

**Pull request merged**